From 1493242620c76ad6057f9dabd3cee6125db06c68 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 28 May 2018 13:28:34 +0800
Subject: [PATCH 4539/5242] MLK-19413-28 drm/imx: dpu: crtc: Export more dpu
 crtc information from dpu-crtc.h

commit  729ea38167cca78cc2d073b07f0be9e0c3343bd3 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch moves several basic structures and helpers of DPU CRTC from
the DPU CRTC driver to dpu-crtc.h so that more DPU CRTC information can
be exported there.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-crtc.c |   44 +--------------------------------
 drivers/gpu/drm/imx/dpu/dpu-crtc.h |   47 +++++++++++++++++++++++++++++++++++-
 2 files changed, 47 insertions(+), 44 deletions(-)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-crtc.c b/drivers/gpu/drm/imx/dpu/dpu-crtc.c
index 7b5f58b..1048a97 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-crtc.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-crtc.c
@@ -26,53 +26,11 @@
 #include <linux/module.h>
 #include <linux/platform_device.h>
 #include <video/dpu.h>
+#include "dpu-crtc.h"
 #include "dpu-kms.h"
 #include "dpu-plane.h"
 #include "imx-drm.h"
 
-struct dpu_crtc {
-	struct device		*dev;
-	struct drm_crtc		base;
-	struct imx_drm_crtc	*imx_crtc;
-	struct dpu_constframe	*cf;
-	struct dpu_disengcfg	*dec;
-	struct dpu_extdst	*ed;
-	struct dpu_framegen	*fg;
-	struct dpu_tcon		*tcon;
-	struct dpu_constframe	*aux_cf;
-	struct dpu_disengcfg	*aux_dec;
-	struct dpu_extdst	*aux_ed;
-	struct dpu_framegen	*aux_fg;
-	struct dpu_tcon		*aux_tcon;
-	struct dpu_plane	**plane;
-	unsigned int		hw_plane_num;
-	unsigned int		stream_id;
-	unsigned int		crtc_grp_id;
-	int			vbl_irq;
-	int			safety_shdld_irq;
-	int			content_shdld_irq;
-	int			dec_shdld_irq;
-
-	struct completion	safety_shdld_done;
-	struct completion	content_shdld_done;
-	struct completion	dec_shdld_done;
-};
-
-struct dpu_crtc_state {
-	struct imx_crtc_state	imx_crtc_state;
-	struct dpu_plane_state	**dpu_plane_states;
-};
-
-static inline struct dpu_crtc_state *to_dpu_crtc_state(struct imx_crtc_state *s)
-{
-	return container_of(s, struct dpu_crtc_state, imx_crtc_state);
-}
-
-static inline struct dpu_crtc *to_dpu_crtc(struct drm_crtc *crtc)
-{
-	return container_of(crtc, struct dpu_crtc, base);
-}
-
 static inline struct dpu_plane_state **
 alloc_dpu_plane_states(struct dpu_crtc *dpu_crtc)
 {
diff --git a/drivers/gpu/drm/imx/dpu/dpu-crtc.h b/drivers/gpu/drm/imx/dpu/dpu-crtc.h
index f624fb7..3896935 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-crtc.h
+++ b/drivers/gpu/drm/imx/dpu/dpu-crtc.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017 NXP
+ * Copyright 2017-2018 NXP
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License as published by the
@@ -15,7 +15,52 @@
 #ifndef _DPU_CRTC_H_
 #define _DPU_CRTC_H_
 
+#include <video/dpu.h>
 #include "dpu-plane.h"
+#include "imx-drm.h"
+
+struct dpu_crtc {
+	struct device		*dev;
+	struct drm_crtc		base;
+	struct imx_drm_crtc	*imx_crtc;
+	struct dpu_constframe	*cf;
+	struct dpu_disengcfg	*dec;
+	struct dpu_extdst	*ed;
+	struct dpu_framegen	*fg;
+	struct dpu_tcon		*tcon;
+	struct dpu_constframe	*aux_cf;
+	struct dpu_disengcfg	*aux_dec;
+	struct dpu_extdst	*aux_ed;
+	struct dpu_framegen	*aux_fg;
+	struct dpu_tcon		*aux_tcon;
+	struct dpu_plane	**plane;
+	unsigned int		hw_plane_num;
+	unsigned int		stream_id;
+	unsigned int		crtc_grp_id;
+	int			vbl_irq;
+	int			safety_shdld_irq;
+	int			content_shdld_irq;
+	int			dec_shdld_irq;
+
+	struct completion	safety_shdld_done;
+	struct completion	content_shdld_done;
+	struct completion	dec_shdld_done;
+};
+
+struct dpu_crtc_state {
+	struct imx_crtc_state	imx_crtc_state;
+	struct dpu_plane_state	**dpu_plane_states;
+};
+
+static inline struct dpu_crtc_state *to_dpu_crtc_state(struct imx_crtc_state *s)
+{
+	return container_of(s, struct dpu_crtc_state, imx_crtc_state);
+}
+
+static inline struct dpu_crtc *to_dpu_crtc(struct drm_crtc *crtc)
+{
+	return container_of(crtc, struct dpu_crtc, base);
+}
 
 struct dpu_plane_state **
 crtc_state_get_dpu_plane_states(struct drm_crtc_state *state);
-- 
1.7.9.5

